src/spatialref.jl

```diff
@@ -616,6 +617,31 @@ function toPROJ4(spref::AbstractSpatialRef)::String
     return unsafe_string(projptr[])
 end
 
+"""
+    toEPSG(spref::AbstractSpatialRef)
+
+If possible, export EPSG code for this coordinate system.
+"""
+function toEPSG(spref::AbstractSpatialRef)::String
+    result = GDAL.osrgetauthorityname(spref.ptr, "PROJCS")
+
+    if !isnothing(result)
+        projcs = "PROJCS"
+    else
+        result = GDAL.osrgetauthorityname(spref.ptr, "GEOGCS")
+        projcs = "GEOGCS"
+    end
+
+    if isnothing(result)
+        error("No PROJCS or GEOGCS Authority found")
+    elseif result == "EPSG"
+            epsg = GDAL.osrgetauthoritycode(spref.ptr, projcs)
+        return "EPSG:$(epsg)"
+    else
+        @ogrerr result "is not an EPSG authority"
```

Comment:
The use of the @ogrerr macro only triggers if GDAL itself failed. Ideally you use this after each call to `osrgetauthoritycode` and `osrgetauthorityname`. Here you probably just want to use error().